tests/docs/testing_framework.md

```diff
@@ -0,0 +1,97 @@
+# An Updated Testing Framework
```

Comment:
is this meant to be a migration doc, or long standing demonstrating the new framework? if the latter probably don't need as much framing around what changed

Reply:
More so a summary of the new framework and what we want to establish going forward in terms of testing structure.  Maybe makes more sense in internal docs?